proposals/trusted_registry_crd.md

```diff
@@ -0,0 +1,128 @@
+# Introduce a custom resource to represent a (trusted) registry
+
+## TOC
+
+- [Abstract](#abstract)
+- [Problem statement](#problem-statement)
+- [Solution](#solutions)
+- [Use cases](#use-cases)
+  - [Starboard Security Operator supervising trusted registries](#starboard-security-operator-supervising-trusted-registries)
+  - [Repository webhooks integrated with Starboard](#repository-webhooks-integrated-with-starboard)
+
+## Abstract
+
+Common requirements for running secure Kubernetes workloads are:
+
+- Pull container images only from trusted registries and/or repositories. For example, one might allow pulling
+  images from private Harbor repository accessible at https://core.harbor.domain/library/nginx:1.16, but deny pull
+  requests to the core.harbor.domain/library/redis:5 or https://docker.io/library/nginx:1.16 repositories.
+- Statically scan container images for critical, high or medium vulnerabilities. In general, these rules are configurable.
+- Use a Kubernetes admission controller to deny workloads with non-compliant container images. For example, one may
+  allow `kubectl run nginx-pod --image=core.harbor.domain/library/nginx:1.16`, but deny
+  `kubectl run redis-pod --image core.harbor.domain/library/redis:5`.
+
+## Problem statement
+
+- Starboard CLI and Starboard Security Operator are able to scan workloads already deployed in a cluster. Hence, an 
+  admission controller cannot use vulnerability reports for compliance checks.
+- The `vulnerabilities.aquasecurity.github.io` resources are scoped to the same namespace as the underlying workload.
+  However, for admission controllers we might want to introduce cluster-scoped resource to represent vulnerability
+  report associated with a given container image digest, for example `registryvulnerabilities.aquasecurity.github.io`.
+- Container images are not first-class citizens in Kubernetes-native world. There’s no built-in resource that represents
+  a container image. Also, there’s no built-in representation of a (trusted) registry.
+- An admission controller has certain performance requirements. Usually we cannot trigger a scanner when the admission
+  controller is invoked for a given Kubernetes API request. Most likely it will take way too long to perform such
+  ad hoc scans.
+
+## Solutions
+
+### Define registries.aquasecurity.github.io resource
```

Comment:
Nit - should this be `registry` in the singular (like `pod` or `node`)? I think this differs from vulnerabilities because we have many vulns in one instance

Reply:
Usually the names that you get from `k api-resources --api-group ""` are plural. Also fully qualified resource names must be plural in CRD descriptors:

```
k explain crd.spec.names.plural
KIND:     CustomResourceDefinition
VERSION:  apiextensions.k8s.io/v1

FIELD:    plural <string>

DESCRIPTION:
     plural is the plural name of the resource to serve. The custom resources
     are served under `/apis/<group>/<version>/.../<plural>`. Must match the
     name of the CustomResourceDefinition (in the form
     `<names.plural>.<group>`). Must be all lowercase
```